### PR TITLE
feat: integrate turborepo

### PR DESCRIPTION
**Pull request by zlatanpham**
Speed up development time by using https://turborepo.org/. It's mostly beneficial for the `build` and `test` command because of the previous build caching mechanism. Below is the benchmark.

## With lerna
```
❯ yarn build
yarn run v1.22.10
$ lerna run build --scope={@sajari/react-components,@sajari/react-hooks,@sajari/react-search-ui,@sajari/react-sdk-utils,@sajari/server}
lerna notice cli v3.22.1
lerna info versioning independent
lerna notice filter including ["@sajari/react-components","@sajari/react-hooks","@sajari/react-search-ui","@sajari/react-sdk-utils","@sajari/server"]
lerna info filter [
lerna info filter   '@sajari/react-components',
lerna info filter   '@sajari/react-hooks',
lerna info filter   '@sajari/react-search-ui',
lerna info filter   '@sajari/react-sdk-utils',
lerna info filter   '@sajari/server'
lerna info filter ]
lerna info Executing command in 5 packages: "yarn run build"
lerna info run Ran npm script 'build' in '@sajari/react-sdk-utils' in 4.5s:
$ tsdx build --tsconfig tsconfig.build.json
✓ Creating entry file 880 ms
✓ Building modules 1.6 secs
lerna info run Ran npm script 'build' in '@sajari/react-hooks' in 7.3s:
$ tsdx build --tsconfig tsconfig.build.json
✓ Creating entry file 1.4 secs
✓ Building modules 4.5 secs
lerna info run Ran npm script 'build' in '@sajari/server' in 3.4s:
$ tsdx build --tsconfig tsconfig.build.json
✓ Creating entry file 953 ms
✓ Building modules 456 ms
lerna info run Ran npm script 'build' in '@sajari/react-components' in 16.1s:
$ tsdx build --tsconfig tsconfig.build.json
✓ Creating entry file 1.4 secs
✓ Building modules 13.4 secs
lerna info run Ran npm script 'build' in '@sajari/react-search-ui' in 7.8s:
$ tsdx build --tsconfig tsconfig.build.json
✓ Creating entry file 1.1 secs
✓ Building modules 5.7 secs
lerna success run Ran npm script 'build' in 5 packages in 28.4s:
lerna success - @sajari/react-components
lerna success - @sajari/react-hooks
lerna success - @sajari/react-search-ui
lerna success - @sajari/server
lerna success - @sajari/react-sdk-utils
✨  Done in 28.88s.
```

### First build
```sh
❯ yarn build
yarn run v1.22.10
$ turbo run build --scope={@sajari/react-components,@sajari/react-hooks,@sajari/react-search-ui,@sajari/react-sdk-utils,@sajari/server}
• Packages in scope: @sajari/react-components, @sajari/react-hooks, @sajari/react-sdk-utils, @sajari/react-search-ui, @sajari/server
• Running build in 5 packages
@sajari/react-sdk-utils:build: cache miss, executing 9d28a33069693a4f
@sajari/react-sdk-utils:build: $ tsdx build --tsconfig tsconfig.build.json
✓ Creating entry file 887 ms
⠸ Building modules  57%                 1.0s, estimated 1.8s
@sajari/react-sdk-utils:build: postcss-nested: postcss.plugin was deprecated. Migration guide:
✓ Building modules 1.7 secs
@sajari/react-hooks:build: cache miss, executing 743f1eb1dbd30cd3
@sajari/react-components:build: cache miss, executing eef1da84db312504
@sajari/react-components:build: $ tsdx build --tsconfig tsconfig.build.json
@sajari/react-hooks:build: $ tsdx build --tsconfig tsconfig.build.json
✓ Creating entry file 1.1 secs
✓ Creating entry file 1.2 secs
@sajari/react-hooks:build: ⠙ Building modules  7%                  0.4s, estimated 5.4s
⠼ Building modules  23%                 1.2s, estimated 5.4s
@sajari/react-components:build: postcss-nested: postcss.plugin was deprecated. Migration guide:
⠏ Building modules  24%                 4.2s, estimated 17.0s
@sajari/server:build: cache miss, executing 8886b2d74b03e463
⠋ Building modules  29%                 4.9s, estimated 17.0s
✓ Creating entry file 929 ms
✓ Building modules 12.5 secs
@sajari/react-search-ui:build: cache miss, executing 09d0a150001c051c
@sajari/react-search-ui:build: $ tsdx build --tsconfig tsconfig.build.json
✓ Creating entry file 1.1 secs
@sajari/react-search-ui:build: ⠙ Building modules  3%                  0.3s, estimated 7.9s
@sajari/react-search-ui:build: postcss-nested: postcss.plugin was deprecated. Migration guide:
✓ Building modules 6 secs

 Tasks:    5 successful, 5 total
Cached:    0 cached, 5 total
  Time:    27.94s 

✨  Done in 28.59s.
```

### Second build with all cache hit
```
❯ yarn build
yarn run v1.22.10
$ turbo run build --scope={@sajari/react-components,@sajari/react-hooks,@sajari/react-search-ui,@sajari/react-sdk-utils,@sajari/server}
• Packages in scope: @sajari/react-components, @sajari/react-hooks, @sajari/react-sdk-utils, @sajari/react-search-ui, @sajari/server
• Running build in 5 packages
@sajari/react-sdk-utils:build: cache hit, replaying output 9d28a33069693a4f
@sajari/react-sdk-utils:build: $ tsdx build --tsconfig tsconfig.build.json
✓ Creating entry file 887 ms
⠸ Building modules  57%                 1.0s, estimated 1.8s
@sajari/react-sdk-utils:build: postcss-nested: postcss.plugin was deprecated. Migration guide:
✓ Building modules 1.7 secs
@sajari/react-hooks:build: cache hit, replaying output 743f1eb1dbd30cd3
@sajari/react-hooks:build: $ tsdx build --tsconfig tsconfig.build.json
✓ Creating entry file 1.1 secs
✓ Building modules 4.1 secs
@sajari/server:build: cache hit, replaying output 8886b2d74b03e463
@sajari/server:build: $ tsdx build --tsconfig tsconfig.build.json
✓ Creating entry file 929 ms
✓ Building modules 399 ms
@sajari/react-components:build: cache hit, replaying output eef1da84db312504
@sajari/react-components:build: $ tsdx build --tsconfig tsconfig.build.json
✓ Creating entry file 1.2 secs
⠸ Building modules  3%                  0.5s, estimated 17.0s
@sajari/react-components:build: postcss-nested: postcss.plugin was deprecated. Migration guide:
✓ Building modules 12.5 secs
@sajari/react-search-ui:build: cache hit, replaying output 09d0a150001c051c
@sajari/react-search-ui:build: $ tsdx build --tsconfig tsconfig.build.json
✓ Creating entry file 1.1 secs
@sajari/react-search-ui:build: ⠙ Building modules  3%                  0.3s, estimated 7.9s
@sajari/react-search-ui:build: postcss-nested: postcss.plugin was deprecated. Migration guide:
✓ Building modules 6 secs

 Tasks:    5 successful, 5 total
Cached:    5 cached, 5 total
  Time:    442ms >>> FULL TURBO

✨  Done in 0.74s.
```

### CODE REVIEW

**Comment by wwalser** [comment]
So, I'm open to moving to turborepo, but honestly the bigger pain point on console at the moment is yarn. Upgrading to Yarn latest is a potential innovation week project for me. Managing deps in console has become very painful because of consistent `workspace` related errors that appear to be known bugs resolved in later versions of Yarn.

Do you have any perf numbers for this change?

**Comment by zlatanpham** [comment]
> Do you have any perf numbers for this change?

Yes. I'm surely going to post the before/after numbers once I can remove the WIP label. I'm still having a bit of trouble with the docs build.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ This will install all dependencies in each project, build them, and symlink them
 In one terminal, run tsdx watch in parallel:
 
 ```sh
-yarn start
+yarn dev
```
Comment (zlatanpham) [inline CONTRIBUTING.md:18]: I changed the dev command from `yarn start` to `yarn dev` since it better represents the purpose.

**Comment by wwalser** [comment]
This has been landed as a part of our migration to modern yarn: https://github.com/sajari/sdk-react/pull/807

**Comment by wwalser** [comment]
This has been landed as a part of our migration to modern yarn: https://github.com/sajari/sdk-react/pull/807

**Pull request closed without merging**